### PR TITLE
Provide support for scraping Azure Network Gateway

### LINKED ISSUE #1264: Provide support for scraping metrics for Azure Network Gateway
Provide support for scraping metrics for Azure Network Gateways

### PR DESCRIPTION
**Pull request by bluepixbe**
Provide support for scraping Azure Network Gateway

Fixes #1264

Metrics output
```
# HELP azure_network_gateway_count_ingress_package_drop Total count of ingress package drops on an Azure network gateway
# TYPE azure_network_gateway_count_ingress_package_drop gauge
azure_network_gateway_count_ingress_package_drop{resource_group="RG",subscription_id="SUB",resource_uri="subscriptions/SUB/resourceGroups/RG/providers/Microsoft.Network/virtualNetworkGateways/Azure-Tele-Gateway",instance_name="Azure-Tele-Gateway"} 19.4 1599219001456
# HELP promitor_ratelimit_arm Indication how many calls are still available before Azure Resource Manager is going to throttle us.
# TYPE promitor_ratelimit_arm gauge
promitor_ratelimit_arm{tenant_id="T",subscription_id="SUB",app_id="APP"} 11996 1599219001431
```

Discovery output
[{"$type":"Promitor.Core.Contracts.ResourceTypes.NetworkGatewayResourceDefinition, Promitor.Core.Contracts","NetworkGatewayName":"Azure-Tele-Gateway","ResourceType":"NetworkGateway","SubscriptionId":"SUB","ResourceGroupName":"RG","ResourceName":"Azure-Tele-Gateway","UniqueName":"Azure-Tele-Gateway"}]

### CODE REVIEW

File: changelog/content/experimental/unreleased.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ version:
  | [#314](https://github.com/tomkerkhove/promitor/issues/314)).
 - {{% tag added %}} Support for scraping Azure Express Route circuits ([docs](https://promitor.io/configuration/v2.x/metrics/express-route-circuit) | [#1251](https://github.com/tomkerkhove/promitor/issues/1251)).
 - {{% tag added %}} Support for scraping Azure Application Gateway ([docs](https://promitor.io/configuration/v2.x/metrics/application-gateway) | [#1251](https://github.com/tomkerkhove/promitor/issues/313)).
+- {{% tag added %}} Support for scraping Azure Network Gateway ([docs](https://promitor.io/configuration/v2.x/metrics/network-gateway) | [#1251](https://github.com/tomkerkhove/promitor/issues/1264)).
```
Comment (tomkerkhove) [inline changelog/content/experimental/unreleased.md:17]:
```suggestion
- {{% tag added %}} Support for scraping Azure Network Gateway ([docs](https://promitor.io/configuration/v2.x/metrics/network-gateway) | [#1264](https://github.com/tomkerkhove/promitor/issues/1264)).
```

File: docs/configuration/v2.x/metrics/network-gateway.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+![Availability Badge](https://img.shields.io/badge/Available%20Starting-v2.0-green.svg)![Resource Discovery Support Badge](https://img.shields.io/badge/Support%20for%20Resource%20Discovery-Yes-green.svg)
+
+You can declare to scrape an Azure Network Gateway via the `NetworkGateway` resource
+type.
+
+The following fields need to be provided:
+
+- `networkGatewayName` - The name of the Azure Network Gateway
+
+All supported metrics are documented in the official [Azure Monitor documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported#microsoftnetworknetworkgateways).
```
Comment (tomkerkhove) [inline docs/configuration/v2.x/metrics/network-gateway.md:17]: This link doesn't work for me, did you mean this one?
```suggestion
All supported metrics are documented in the official [Azure Monitor documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported#microsoftnetworkvirtualnetworkgateways).
```

**Review by tomkerkhove — COMMENTED** [review]
LGTM, thanks for your contribution!

I've added a note on (https://github.com/tomkerkhove/promitor/issues/1264#issuecomment-688073057) with regards to the naming of the resource type given it's new to me and want to check what the best name is and/or we need to introduce an alias for VpnGateway.

**Review by tomkerkhove — APPROVED** [review]

File: changelog/content/experimental/unreleased.md [hunk 3]
```diff
@@ -14,19 +14,21 @@ version:
  | [#314](https://github.com/tomkerkhove/promitor/issues/314)).
 - {{% tag added %}} Support for scraping Azure Express Route circuits ([docs](https://promitor.io/configuration/v2.x/metrics/express-route-circuit) | [#1251](https://github.com/tomkerkhove/promitor/issues/1251)).
 - {{% tag added %}} Support for scraping Azure Application Gateway ([docs](https://promitor.io/configuration/v2.x/metrics/application-gateway) | [#1251](https://github.com/tomkerkhove/promitor/issues/313)).
```
[the remark this answers is not in the document]
Comment (tomkerkhove) [reply changelog/content/experimental/unreleased.md:16]: It's ok

**Comment by tomkerkhove** [comment]
Thanks so much @bluepixbe!

**Comment by bluepixbe** [comment]
@tomkerkhove cool thanks for the merge.

**Pull request merged**